src-plugins/vtkDataMesh/readers/vtkDataMeshReader.cpp

```diff
@@ -89,28 +90,139 @@ bool vtkDataMeshReader::read(const QString& path) {
             return false;
         }
 
-        try
+        // Get the extension of the filename
+        QFileInfo pathfile(path);
+        QString extension = pathfile.completeSuffix();
+
+        if (extension == "vtk") // VTK files
+        {
+            // Extract data and header from the file
+            QString header;
+            vtkSmartPointer<vtkPolyDataReader> reader = vtkPolyDataReader::New();
+            try
+            {
+                reader->SetFileName(path.toLocal8Bit().constData());
+                reader->Update();
+
+                header = reader->GetHeader();
+                dataSet->Read(path.toLocal8Bit().constData());
+            }
+            catch (vtkErrorCode::ErrorIds error)
+            {
+                qDebug() << "vtkDataMeshReader: " << vtkErrorCode::GetStringFromErrorCode(error);
+                return false;
+            }
+
+            medData->setData(dataSet);
+
+            // Parse header and save metadata in medData
+            parseHeaderVtk(header, medData);
+        }
+        else if (extension == "vtp") // VTP files
         {
             dataSet->Read(path.toLocal8Bit().constData());
-        } catch (...)
+            medData->setData(dataSet);
+
+            // Extract field data from the xml file
+            vtkSmartPointer<vtkXMLPolyDataReader> reader = vtkSmartPointer<vtkXMLPolyDataReader>::New();
+            reader->SetFileName(path.toLocal8Bit().constData());
+            reader->Update();
+
+            if (reader->GetOutput()->GetFieldData())
+            {
+                parseHeaderVtp(reader->GetOutput()->GetFieldData(), medData);
+            }
+        }
+        else if ((extension == "mesh") || (extension == "obj")) // MESH or OBJ files
+        {
+            try
+            {
+                dataSet->Read(path.toLocal8Bit().constData());
+            } catch (...)
+            {
+                qDebug() << "vtkDataMeshReader::read -> loading the vtkDataMesh failed, error while parsing !";
```

Comment:
This catch block will be called for ANY exception, so it's not necessarily a parsing error. The error message should take into account all possibilities. Something like "error while trying to read [filename]".